main.go

```diff
@@ -263,23 +262,40 @@ func main() {
 	}
 
 	// Configure metrics
-	metricsHost := os.Getenv(statsdHostVariable)
-	metricsPort := os.Getenv(statsdPortVariable)
-	metricsTimeout, err := strconv.ParseInt(os.Getenv(statsdTimeoutVariable), 10, 64)
-	if err != nil {
-		log.Printf("Failed parsing metrics timeout: %s", err)
-		metricsTimeout = 100
-	}
-	client, err := createStatsDClient(metricsHost, metricsPort, int(metricsTimeout))
-	if err != nil {
-		log.Fatalf("Failed to create statsd client: %s", err)
-	}
-	defer client.Close()
+	var metricsFactory MetricsFactory
+
+	if prometheusConfigJSON := os.Getenv(prometheusConfigVariable); prometheusConfigJSON != "" {
+		log.Printf("prometheus config: %s", prometheusConfigJSON)
+		var prometheusConfig PrometheusConfig
+		if err := json.Unmarshal([]byte(prometheusConfigJSON), &prometheusConfig); err != nil {
+			log.Fatalf("Failed to parse Prometheus config: %s", err)
+		}
+
+		metricsFactory, err = NewPrometheusMetricsFactory(prometheusConfig)
+		if err != nil {
+			log.Fatalf("Failed to configure Prometheus metrics: %s", err)
+		}
+	} else {
+		// Default to StatsD metrics
+		monitoringServiceName := getStringEnv(monitoringServiceNameEnvironmentVariable, defaultMonitoringServiceName)
+		metricsHost := os.Getenv(statsdHostVariable)
+		metricsPort := os.Getenv(statsdPortVariable)
+		metricsTimeout, err := strconv.ParseInt(getStringEnv(statsdTimeoutVariable, "100"), 10, 64)
```

Comment:
There was a bug here where we used to use `strconv.ParseInt(os.Getenv(statsdTimeoutVariable))`, and if one wasn't using StatsD metrics, you'd always get an annoying "Failed parsing metrics timeout:" logline. Now we use `getStringEnv` and default to 100, which matches the existing behavior without nagging operators.